src/Geohash/GeohashInterface.php

```diff
@@ -24,13 +26,13 @@ interface GeohashInterface
      * Returns a geo hash string.
      *
      * @param CoordinateInterface $coordinate The coordinate to encode.
-     * @param integer             $length     The length of the hash between 1 to 12 by default (optional).
+     * @param int                 $length     The length of the hash between 1 and 12 by default (optional).
      *
      * @return GeohashInterface
      *
      * @throws InvalidArgumentException
      */
-    public function encode(CoordinateInterface $coordinate, $length = self::MAX_LENGTH);
```

Comment:
Why did you remove the default value of `$length`? It's a breaking change for the GeohashInterface.

Reply:
This file refers to an Interface, it's not the class itself. Therefore, `self::MAX_LENGTH` cannot be resolve at this place, as `self::` refers to nothing, and `public const MAX_LENGTH = 12;` is only defined in the Geohash class, not in the Interface

Reply:
The default value is still present in the `Geohash::encore` method definition

Reply:
Exact, I didn't notice the default value in `Geohash::encode`. You could remove `(optional)` in the comment above.